app/livechat/server/lib/Livechat.js

```diff
@@ -1119,6 +1119,39 @@ export const Livechat = {
 
 		return Promise.await(businessHourManager.allowAgentChangeServiceStatus(agentId));
 	},
+
+	notifyRoomVisitorChange(roomId, visitor) {
+		Livechat.stream.emit(roomId, {
+			type: 'visitorData',
+			visitor,
+		});
+	},
+
+	changeRoomVisitor(userId, roomId, visitor) {
+		const user = Promise.await(Users.findOneById(userId));
+		if (!user) {
+			throw new Error('error-user-not-found');
+		}
+
+		if (!hasPermission(userId, 'change-livechat-room-visitor')) {
+			throw new Error('error-not-authorized');
+		}
+
+		const room = Promise.await(LivechatRooms.findOneById(roomId));
+		if (!room) {
+			throw new Meteor.Error('invalid-room-id');
+		}
+
+		if (!canAccessRoom(room, user)) {
+			throw new Error('error-not-allowed');
+		}
+
+		LivechatRooms.changeVisitorByRoomId(room._id, visitor);
+
+		Livechat.notifyRoomVisitorChange(room._id, visitor);
+
+		return room;
```

Comment:
You need to return the updated room object after changing the room visitor.

Reply:
I've modified the method to return the room object